test/e2e/fixture/fixture.go

```diff
@@ -196,6 +183,42 @@ func init() {
 
 }
 
+func LoginAsAdmin() {
+	closer, client, err := ArgoCDClientset.NewSessionClient()
+	CheckError(err)
+	defer io.Close(closer)
+
+	SessionClient = client
+
+	sessionResponse, err := client.Create(context.Background(), &sessionpkg.SessionCreateRequest{Username: adminUsername, Password: adminPassword})
+	CheckError(err)
+	token = sessionResponse.Token
+
+	ArgoCDClientset, err = argocdclient.NewClient(&argocdclient.ClientOptions{
+		Insecure:   true,
+		ServerAddr: apiServerAddress,
+		AuthToken:  token,
+		PlainText:  plainText,
+	})
+	CheckError(err)
+}
+
+func LoginAs(username string) {
+	if username == "admin" {
+		LoginAsAdmin()
+		return
+	}
+
+	var err error
+	token, err = RunCli("account", "generate-token", "--account", username)
```

Comment:
The `generate-token` creates an entry in `argocd-secret`. We might end up having a lot of keys at the end of e2e tests. To avoid it I would suggest to create accounts with hard coded passwords (e.g. `password`) and use `Session.Create` grpc to create token that is not persisted. This way we can use the same implementation for admin and other users.

Reply:
Thank you for review. I think we remove all these tokens during clean process that we run inside each .Given(), but I still agree with you that such approach is better, actually I pushed a fix :)